lib/blacklight/configuration.rb

```diff
@@ -65,6 +65,7 @@ def default_values
           index: ViewConfig::Index.new(
             # document presenter class used by helpers and views
             document_presenter_class: nil,
+            document_component: nil,
```

Comment:
could we have the default be `Blacklight::DocumentComponent` and avoid a nil check in the view?

Reply:
I made it `nil` so we'd have some way to know that an implementation has opted in to the view components. Other ideas?

Reply:
Please add a comment that explains that anything other than nil indicates the implementation has opted in to components.